lib/archethic/reward.ex

```diff
@@ -53,7 +53,17 @@ defmodule Archethic.Reward do
     trunc(50 / uco_usd_price / number_of_reward_occurences_per_month * @unit_uco)
   end
 
-  defp number_of_reward_occurences_per_month(), do: @number_of_reward_occurences_per_month
+  defp number_of_reward_occurences_per_month() do
```

Comment:
If I understand correctly, the function `Utils.number_of_reward_occurences_per_month()` is working well and the issue is that it was called only at compile time.

Isn't it possible to just move it to runtime like this ?

```
defp number_of_reward_occurences_per_month() do 
    datetime = NaiveDateTime.utc_now()
    interval = Application.get_env(:archethic, RewardScheduler)[:interval]
    Utils.number_of_reward_occurences_per_month(interval, datetime)
end
```

Reply:
no because in some cases depending on the interval  it can take more than 1s to execute which is not acceptable